incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/MqttKafkaConfiguration.java

```diff
@@ -21,11 +21,15 @@ public class MqttKafkaConfiguration extends Configuration
     private static final ConfigurationDef MQTT_KAFKA_CONFIG;
 
     public static final PropertyDef<String> KAFKA_MESSAGES_TOPIC;
+    public static final PropertyDef<String> KAFKA_RETAINED_MESSAGES_TOPIC;
+    public static final BooleanPropertyDef RETAIN_AVAILABLE;
 
     static
     {
         final ConfigurationDef config = new ConfigurationDef("zilla.binding.mqtt.kafka");
         KAFKA_MESSAGES_TOPIC = config.property("messages.topic", "mqtt_messages");
+        KAFKA_RETAINED_MESSAGES_TOPIC = config.property("retained.messages.topic", "mqtt_retained");
+        RETAIN_AVAILABLE = config.property("retain.available", true);
```

Comment:
I added this so that we won't see retain streams in non-retain related scenarios in the scripts: like for sessions in the mqtt binding. But in reality, this is always true (if we don't specify it from the test). 
Since it's always true in prod, in reality we're responding to the MqttPublish and MqttSubscribe streams flags.
Let me know if you think we should delete it and adapt the scenarios for the change.

Reply:
I think we should be responding to the flags in the beginEx only and not have separate configuration. 

The scenarios should be kept simple (without retained as input) and therefore not send a retained stream.

Reply:
Sounds good for subscribe retained, but I think we discussed that the publish retained stream is always open, and if the client did not send any retained message, we just keep that stream in sync

Reply:
Understood. Is there configuration in `mqtt` binding to disable `retained` publishing?

Reply:
For mqtt publish beginEx add a flag: retain available or not. If not we can assume that we won't get publish with retain flag and can reject them if we do. Honor the retain flag in the subscribe stream beginEx